HUDOC.js

```diff
@@ -0,0 +1,502 @@
+			if (elem.includes("V.") || elem.includes("C.")) {
+						return true;
+			}
+			return false;
+		});
+
+		let titleNumMatch = splitTitle.findIndex(function(elem) {
+			if (/^(no.|II|IV|V\)|VI|IX|X\))/i.test(elem)) {
+						return true;
+			}
+			return false;
+		});
+
+		let titleIndex = Math.max(titleVersusMatch, titleNumMatch);
+
+
+		capTitle = splitTitle.slice(0, titleIndex+1).join("(");
+	}
+
+	// Zotero capitalizes the "v."/"c.", so we need to correct that for English and French cases
+	if (capTitle.includes(" V. ")) {
+		capTitle = capTitle.replace(" V. ", " v. ");
+	}
+	else if (capTitle.includes(" C. ")) {
+		capTitle = capTitle.replace(" C. ", " c. ");
+	}
+	item.caseName = capTitle + getTypeBit(doc, url);
+
+	// Court
+	var court = scrapeMetaData(doc, "originatingbody");
+
+	if (url.includes("hudoc.echr.coe.int/fre#")) {
+		item.court = "ECtHR";
+	}
+	else {
+		if (text(doc, "title").includes("Advisory Opinion")) {
+			item.court = "ECtHR [GC]";
+		}
+		else if (court.includes("Grand Chamber")) {
+			item.court = "ECtHR [GC]";
+		}
+		else if (court.includes("Commission")) {
+			item.court = "Commission";
+		}
+		else if (court == "Committee of Ministers") {
+			item.court = "Committee of Ministers";
+		}
+		else {
+			item.court = "ECtHR";
+		}
+	}
+
+	// Date of decision
+	// convert to simple ISO: yyyy-mm-dd dd.mm.yyyy.
+	item.dateDecided = scrapeMetaData(doc, "judgementdate").split("/").reverse().join('-');
+
+	item.url = "http://hudoc.echr.coe.int/eng?i=" + getItemID(url);
+
+	// Query remaining metadata from API
+	var queryUrl = "https://hudoc.echr.coe.int/app/query/results?query=(contentsitename=ECHR) AND "
+					+ getItemID(url)
+					+ "&select=appno,conclusion,docname,languageisocode" // Adapt based on what is needed
+					+ "&sort=&start=0&length=1";
+
+	ZU.doGet(queryUrl, function (json) {
+		json = JSON.parse(json).results[0].columns;
+
+		Zotero.debug("Queried HUDOC API at: " + queryUrl);
+
+		// Application numbers
+		// FIXME: Decide whether to prepend "app. no.". Some styles automatically add "no." (Chicago)
+		// while most others do not (OSCOLA, APA).
+		var appno = json.appno.split(";");
+		if (appno.toString().length !== 0) { // Only if not empty
+			if (appno.length > 1) {
+				var length = appno.length;
+				var commaPart = appno.slice(0, length - 1);
+				var appnos = commaPart.join(", ") + " and " + appno[length - 1].toString();
+
+				item.docketNumber = "app. nos. " + appnos;
+			} else {
+				item.docketNumber = "app. no. " + appno;
+			}
+		}
+
+		// Abstract
+		item.abstractNote = json.conclusion.replace(/;/g, "; ");
+
+		// Language
+		item.language = json.languageisocode.toLowerCase();
+
+		// Download PDF
+		var docname = json.docname;
+
+		var pdfurl = "https://hudoc.echr.coe.int/app/conversion/docx/pdf?library=ECHR&id="
+					+ getItemID(url) + "&filename=" + docname + ".pdf";
+
+		pdfurl = encodeURI(pdfurl); // the "docname" may contain chars not part of the URI schema
```

Comment:
Rather than encoding the entire URL, try calling `encodeURIComponent` on just `docname` before adding it to the URL. That will escape `&` and other characters that don't belong.

Reply:
Fixed